mirrord/agent/src/main.rs

```diff
@@ -234,7 +234,16 @@ impl ClientConnectionHandler {
         let (tcp_sender, tcp_receiver) = mpsc::channel(CHANNEL_SIZE);
 
         let tcp_sniffer_api =
-            TcpSnifferApi::new(id, sniffer_command_sender, tcp_receiver, tcp_sender).await?;
+            TcpSnifferApi::new(id, sniffer_command_sender, tcp_receiver, tcp_sender)
+                .await
+                .map_err(|e| {
+                    warn!(
+                        "Failed to create TcpSnifferApi: {}, this could be due to kernel version.",
+                        e
+                    );
+                    e
```

Comment:
Nitpick: I think you could use `inspect_err` here instead of returning the same error.

Reply:
I don't like adding more dependencies on nightly